source/load-balancers/load-balancers.rst

```diff
@@ -0,0 +1,531 @@
+
+It was decided that the core problem that was needed to be solved was to introduce durable/resumable
+cursors to some services. The related ticket is `DRIVERS-1549 <https://jira.mongodb.org/browse/DRIVERS-1549>`__.
+
+
+Design Rationales
+-----------------
+
+This design is based on the simplest approach to implementation in both the drivers and all services,
+when taking into consideration that it is NOT a requirement that applications connected to any service
+should not need to make changes and/or restart their applications. If this requirement were to change,
+see alternative designs below.
+
+Connection pinning behaviour is not altered in this specification and it is understood that this can
+result in connections being pinned longer than necessary in some cases. This could result in session
+and cursor leaks resulting in connection leaks but this was deemed as acceptable in this case.
+
+
+Alternative Designs
+-------------------
+
+Service PROXY Detection
+^^^^^^^^^^^^^^^^^^^^^^^
+
+An alternative to the driver using a connection string option to put it into load balancing
+mode would be for the service the driver is connected to to inform the driver it is behind
+a load balancer. A possible solution for this would be for all services to understand the
+PROXY protocol such as Data Lake does, and to alter their hello responses to inform the
+driver they are behind a load balancer, potentially with the IP address of the load balancer itself.
+
+The benefit of this solution would be that no changes would be required from the application
+side, and could also not require a restart of any application. A single request to the service
+through the load balancer could automatically trigger the change in the hello response and
+cause the driver to switch into load balancing mode pointing at the load balancer's IP address.
+Also with this solution it would provide services the ability to record the original IP addresses
+of the application that was connecting to it as they are provided the PROXY protocol's header bytes.
+
+The additional complexity of this alternative on the driver side is that instead of starting
+in a single mode and remaining there for the life of the application, the driver would need
+to deal with additional state changes based on the results of the server monitors. From a
+service perspective, every service would need to be updated to understand the PROXY protocol
+header bytes prepended to the initial connection and modify their states and hello responses
+accordingly. Additionally load balancers would need to have additional configuration as noted
+in the reference section below, and only load balancers that support the PROXY protocol would
+be supported.
+
+
+References
+----------
+
+Data Lake PROXY Detection
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+`Github Link <https://github.com/10gen/mongohouse/blob/master/internal/util/proxy/conn.go#L108>`__
+
+Load Balancers PROXY Protocol Configuration
```

Comment:
I think we can remove this whole section as well.

Reply:
Removed.